### ISSUE TITLE
Adapt InternalEngineTests#testUpgradeOldIndex for 5.0.0 BWC indices

### ISSUE THREAD

**Issue by jasontedor** [elastic/elasticsearch#21147]
This test is not ready to handle 5.0.0 BWC indices. Since this test uses randomization to select indices to test, a seed that causes a 5.0.0 BWC to be selected is `2CDB18C976529809`.

**Comment by jasontedor** [comment]
I pushed 92788882f23a6149095fac49b0386049567e4fc4 to silence this test for now.

**Comment by s1monw** [comment]
I fixed this on the 5.0 branch but on 5.x and master I am getting:
```REPRODUCE WITH: gradle null -Dtests.seed=B3255701D6550F90 -Dtests.class=org.elasticsearch.index.engine.InternalEngineTests -Dtests.method="testUpgradeOldIndex" -Dtests.locale=es-PY -Dtests.timezone=Pacific/Bougainville

java.lang.RuntimeException: packed points cell maxPackedValue [-128, 0, 0, 0, 0, 0, 0, 0] is out-of-bounds of the global minimum [-128, 0, 0, 0, 0, 0, 0, 0] dim=0

	at __randomizedtesting.SeedInfo.seed([B3255701D6550F90:CD165694FAB6F103]:0)
	at org.apache.lucene.index.CheckIndex$1.compare(CheckIndex.java:1908)
	at org.apache.lucene.util.bkd.BKDReader.intersect(BKDReader.java:432)
	at org.apache.lucene.util.bkd.BKDReader.intersect(BKDReader.java:476)
	at org.apache.lucene.util.bkd.BKDReader.intersect(BKDReader.java:285)
	at org.apache.lucene.codecs.lucene60.Lucene60PointsReader.intersect(Lucene60PointsReader.java:142)
	at org.apache.lucene.index.CheckIndex.testPoints(CheckIndex.java:1841)
	at org.apache.lucene.index.CheckIndex.checkIndex(CheckIndex.java:753)
	at org.apache.lucene.util.TestUtil.checkIndex(TestUtil.java:300)
	at org.apache.lucene.store.MockDirectoryWrapper.close(MockDirectoryWrapper.java:865)
	at org.apache.lucene.store.FilterDirectory.close(FilterDirectory.java:109)
	at org.elasticsearch.index.store.Store$StoreDirectory.innerClose(Store.java:682)
	at org.elasticsearch.index.store.Store$StoreDirectory.access$100(Store.java:656)
	at org.elasticsearch.index.store.Store.closeInternal(Store.java:364)
	at org.elasticsearch.index.store.Store.access$000(Store.java:126)
	at org.elasticsearch.index.store.Store$1.closeInternal(Store.java:147)
	at org.elasticsearch.common.util.concurrent.AbstractRefCounted.decRef(AbstractRefCounted.java:64)
	at org.elasticsearch.index.store.Store.decRef(Store.java:348)
	at org.elasticsearch.index.store.Store.close(Store.java:356)
	at org.apache.lucene.util.IOUtils.close(IOUtils.java:89)
	at org.apache.lucene.util.IOUtils.close(IOUtils.java:76)
	at org.elasticsearch.index.engine.InternalEngineTests.testUpgradeOldIndex(InternalEngineTests.java:1905)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)```

which seems like a lucene bug in the 6.3.0 snapshot bulid we are using. @mikemccand is investigating

**Comment by jasontedor** [comment]
Thanks @s1monw and @mikemccand.

**Comment by s1monw** [comment]
this is enabled again on 5.0 but not on 5.x and master until we get a new build. @jpountz if you happen to upgrade to a new snap can you remove the `@AwaitsFix` from this test on the relevant branches please.

**Comment by s1monw** [comment]
fixed in 0a556acf7931c3f5b7785d52dd5c2f7577b74754
